src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/ViewExecutor.cs

```diff
@@ -84,12 +122,27 @@ public static class ViewExecutor
                     viewData,
                     tempData,
                     writer,
-                    htmlHelperOptions);
+                    ViewOptions.HtmlHelperOptions);
+
+                if (Telemetry.IsEnabled("Microsoft.AspNet.Mvc.BeforeView"))
+                {
+                    Telemetry.WriteTelemetry(
+                        "Microsoft.AspNet.Mvc.BeforeView",
+                        new { view = view, viewContext = viewContext, });
+                }
 
                 await view.RenderAsync(viewContext);
-                // Invoke FlushAsync to ensure any buffered content is asynchronously written to the underlying
-                // response. In the absence of this line, the buffer gets synchronously written to the response
-                // as part of the Dispose which has a perf impact.
+
+                if (Telemetry.IsEnabled("Microsoft.AspNet.Mvc.AfterView"))
+                {
+                    Telemetry.WriteTelemetry(
+                        "Microsoft.AspNet.Mvc.AfterView",
+                        new { view = view, viewContext = viewContext, });
+                }
+
```

Comment:
Just how many messages I need to processes to determine if the timing I have (between BeforeView and AfterView) was for a partial view or not - we show them differently in the ui.